File: Pcb.WebNg/src/app/models/recipe.ts
import { ApiLinks } from './apiLinks';
import { History } from './history';
import { IngredientList } from './ingredientList';
import { Picture } from './picture';
import { RatingReviews } from './ratingsReview';
import { RecipeCreatedByUser } from './recipeCreatedByUser';
import { StepByStepInstructions } from './stepByStepInstructions';


export interface Recipe {
	id?: number;
	/**
	 * unique name of the recipe
	 */
	name: string;
	/**
	 * short text description
	 */
	teaser?: string;
	/**
	 * Number of people or servings the recipe will make
	 */
	numberOfServings: number;
	/**
	 * calculated field from average prices
	 */
	priceEstimate?: number;
	/**
	 * amount of time in minutes to prepare everything estimate
	 */
	prepTime?: number;
	/**
	 * amount of time in minutes to cook the recipe
	 */
	cookTime?: number;
	/**
	 * amount of time in minutes to cook the recipe
	 */
	readyInMinutes?: number;
	/**
	 * calculated field of price / number of servings
	 */
	priceServing?: number;
	/**
	 * string can be very long it appears
	 */
	rawInstructions: string;
	createByUserId: number;
	/**
	 * eg CWA Cookbook 2015 or student/teacher/import created or website url
	 */
	sourceOfRecipeName?: string;
	sourceOfRecipeLink?: string;
	spoonacularId?: string;
	creditsText?: string;
	numberStars?: number;
	numberReviews?: number;
	numberFavourites?: number;
	numberOfTimesCooked?: number;


	equipmentRequired?: Array<string>;

	recipeAllergyWarning?: Array<string>;
	/**
	 * mexican, chinese, european, tudor, etc
	 */
	recipeCuisineType?: Array<string>;

	/** Very healthy, cheap, popular, sustainable, etc */
	recipeDishTags?: Array<string>;
	/**
	 * more than one type possible - breakfast, lunch, dinner snack, sauce, base
	 */
	recipeDishType?: Array<string>;
	/**
	 * low sugar, paleo, sugar free
	 */
	recipeHealthLabels?: Array<string>;
	/** analysed Instructions broken down to steps */
	steppedInstructions?: Array<StepByStepInstructions>;
	recipeIngredientLists?: Array<IngredientList>;
	recipePicture?: Array<Picture>;
	recipeReview?: Array<RatingReviews>;

	createdByUser?: RecipeCreatedByUser;


	/**
	 * all the events that occur for the recipe - ie created, edited, booked etc
	 */
	history?: Array<History>;
	links?: ApiLinks;
	updatedAt?: string | Date;
	createdAt?: string | Date;
	rowVer?: string;
}

File: Pcb.Database/Context/Models/Recipe.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;

namespace Pcb.Database.Context.Models
{
    [Table("Recipe", Schema = "dbo")]
    public partial class Recipe
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Teaser { get; set; }
        public int NumberOfServings { get; set; }
        [DataType(DataType.Currency)]
        [Column(TypeName = "decimal(18,2)")]

        public decimal PriceEstimate { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        [DataType(DataType.Currency)]
        public decimal PriceServing { get; set; }
        public int PrepTime { get; set; }
        public int CookTime { get; set; }
        public int ReadyInMinutes { get; set; }
        [Column(TypeName = "text")]
        public string RawInstructions { get; set; }
        public int CreateByUserId { get; set; }
        public string SourceOfRecipeName { get; set; } // used to reference source name
        public string SourceOfRecipeLink { get; set; } // used to reference SpoonAcular or source
        public string SpoonacularId { get; set; }
        public string CreditsText { get; set; }
        public int NumberStars { get; set; }
        public int NumberFavourites { get; set; }
        public int NumberOfTimesCooked { get; set; }

        [Timestamp]
        public byte[] RowVer { get; set; }

        public DateTimeOffset CreatedAt { get; set; }

        public ICollection<RecipeAllergyWarning> RecipeAllergyWarning { get; set; }
        public ICollection<RecipeCuisineType> RecipeCuisineType { get; set; }
        public ICollection<RecipeDishTag> RecipeDishTag { get; set; }
        public ICollection<RecipeDishType> RecipeDishType { get; set; }
        public ICollection<RecipeHealthLabel> RecipeHealthLabel { get; set; }
        public ICollection<RecipeSteppedInstruction> SteppedInstruction { get; set; }
        public ICollection<RecipeIngredientList> RecipeIngredientList { get; set; }
        public ICollection<RecipeReview> RecipeReview { get; set; }
        public ICollection<RecipePicture> RecipePicture { get; set; }
    }
}
